docs/BUILD

```diff
@@ -102,8 +110,84 @@ genrule(
 genrule(
     name = "examples_rst",
     outs = ["examples_rst.tar"],
-    cmd = "tar cfh $@ --transform='s/examples/start\\/sandboxes\\/_include/' $(locations //examples:files)",
+    cmd = """
+    tar cfh $@ --transform='s/examples/start\\/sandboxes\\/_include/' \\
+        $(locations //examples:files)
+    """,
+    tools = ["//examples:files"],
+)
+
+genrule(
+    name = "extensions_security_rst",
+    srcs = ["//source/extensions:extensions_metadata.yaml"],
+    outs = ["extensions_security_rst.tar"],
+    cmd = """
+    $(location //tools/docs:generate_extensions_security_rst) \\
+        $(location //source/extensions:extensions_metadata.yaml) $@
+    """,
+    tools = ["//tools/docs:generate_extensions_security_rst"],
+)
+
+genrule(
+    name = "external_deps_rst",
+    srcs = [
+        "//bazel:repository_locations.bzl",
+        "@envoy_api_canonical//bazel:repository_locations.bzl",
+        "@envoy_api_canonical//bazel:repository_locations_utils.bzl",
+    ],
+    outs = ["external_deps_rst.tar"],
+    cmd = "$(location //tools/docs:generate_external_deps_rst) $@",
+    tools = ["//tools/docs:generate_external_deps_rst"],
+)
+
+genquery(
+    name = "v3_proto_srcs",
+    expression = "labels(srcs, labels(deps, @envoy_api_canonical//:v3_protos))",
+    scope = ["@envoy_api_canonical//:v3_protos"],
+)
+
+genrule(
+    name = "empty_protos_rst",
+    srcs = [":empty_extensions.json"],
+    outs = ["empty_protos_rst.tar"],
+    cmd = """
+    $(location //tools/protodoc:generate_empty) \\
+        $(location empty_extensions.json) $@
+    """,
+    tools = ["//tools/protodoc:generate_empty"],
+)
+
+genrule(
+    name = "api_rst",
+    srcs = [
+        "//tools/protodoc:api_v3_protodoc",
+        ":v3_proto_srcs",
+    ],
+    outs = ["api_rst.tar"],
+    cmd = """
+    $(location //tools/docs:generate_api_rst) \\
+        $(location v3_proto_srcs) $(locations //tools/protodoc:api_v3_protodoc) $@
+    """,
+    tools = ["//tools/docs:generate_api_rst"],
+)
+
+genrule(
+    name = "rst",
+    outs = ["rst.tar"],
+    cmd = """
+    cat $(location base_rst) > $@ \\
```

Comment:
FWIW, I think this can get a bit silly if we keep pushing more and more stuff directly into `genrule` (and some multiplicity of them in a graph). I'd advocate for clarity to try and use a shell script where we don't need parallelism and things are a bit more readable. I'll approve this PR and leave that to your taste.

Reply:
mho is that its confusing having a mix of bazel rules and shell scripts - compounded by using env vars to pass around data

i believe doing it this way we can use rbe and speed everything up (i hope at least)

Reply:
also the bazel docs suggest this iirc (i would need to trudge back through to find the ref) - something about using lots of locations to pass around the built data